backend/main.py

```diff
@@ -0,0 +1,24 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+from routers import api_endpoints
+
+app = FastAPI(title="OpenSearch Automation API")
+
+# Configure CORS
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["http://localhost:3000"],  # React frontend URL
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+# Include routers
+app.include_router(api_endpoints.router, prefix="/api/cluster", tags=["cluster"])
+
+if __name__ == "__main__":
+    import uvicorn
+    uvicorn.run(app, host="0.0.0.0", port=8000) 
```

Comment:
Binding to 0.0.0.0 exposes the server to all network interfaces. For development, use 127.0.0.1. Consider making this configurable via environment variables.